src/test/java/org/opensearch/neuralsearch/processor/TextChunkingProcessorTests.java

```diff
@@ -630,15 +649,18 @@ public void testExecute_withFixedTokenLength_andFieldMapNestedMap_sourceDataList
     }
 
     @SneakyThrows
-    public void testExecute_withFixedTokenLength_andSourceDataListWithHybridType_thenSucceed() {
+    public void testExecute_withFixedTokenLength_andSourceDataListWithHybridType_thenFail() {
```

Comment:
Why this test case should fail?

Reply:
Current code we didn't have bi-directional validation of document source and configuration map. A map type configuration can't fit both string and map type value, e.g. 
```
{
  "field_map": {
    "a": {
      "b": {
        "c": "c_target"
      }
    }
  }
}
```
Can not fit to both(case 1)
```
{
  "a": {
    "b": {
      "c": "text to embedding/chunk"
    }
  }
}
```
and(case 2)
{
  "a": {
    "b": "text to embedding/chunk"
  }
}
```
For case 2, text_embedding and text_image_embedding processor needs to build a temporary map with target key, and `b`'s type will be treated as map based on configuration and then cause class cast exception. 
For text chunking, current code doesn't throw exception but the result is not correct, based on user's configuration, `c` is the target chunking field but the actual chunking field in this case is `b`, and the generated result is:
```
{
  "c_target": "b's chunking result"
}
```
Which is unexpected.

So now we enforced the bi-directional to avoid this case, and the base principle we'll follow in the future is: list type always have same type/data structure object in it.

Reply:
Agree with `list type always have same type/data structure object in it`. However, it maybe a breaking change for not supporting hybrid list. Users previously pass in hybrid input in a list will now receive an error. Can we avoid throwing an error the user?

Reply:
Breaking change means an expected behavior changed, like above mentioned, this in fact is a bug since the result is not what user expected, by adding this validation user can realize the issue and fix their inputs.